rapid/src/main/java/com/vrg/rapid/MembershipService.java

```diff
@@ -298,21 +299,28 @@
         final SettableFuture<RapidResponse> future = SettableFuture.create();
 
         sharedResources.getProtocolExecutor().execute(() -> {
+            // first, retain all valid alerts
+            // this includes adding the UUIDs and metadata of joining nodes
+            final long currentConfigurationId = membershipView.getCurrentConfigurationId();
+            final int membershipSize = membershipView.getMembershipSize();
+            final Stream<AlertMessage> validAlerts = messageBatch.getMessagesList().stream()
+                // First, we filter out invalid messages that violate membership invariants.
+                .filter(msg -> filterAlertMessagesAndAddJoinerDetails(messageBatch, msg, membershipSize,
```

Comment:
It's worth adding a comment here stating that this step is stateful and needs to be done regardless of whether we have announced a proposal or not.